rfd/0086/mariposa.md

```diff
@@ -0,0 +1,73 @@
+## Mariposa integration
+
+There are a number of cases where we've wanted for lifecycle hooks to be able to communicate information back to ContainerPilot, or to be otherwise able to alter the ContainerPilot environment for purposes of future handler events or the main application itself. Additionally, it's likely that we at Joyent will want the [RFD36](https://github.com/joyent/rfd/blob/master/rfd/0036/README.md) scheduler for Triton to be able to have an interface to communicate with containers and ContainerPilot itself, for which we can provide an example in ContainerPilot.
+
+This control plane would be useful for event hooks as well. Telemetry sensor outputs will be consumed via a robust API instead of the existing brittle text scraping. Event hooks will also be able to set environment variables for other services and event hooks.
+
+This proposal is to create an HTTP server for ContainerPilot that exposes a simple HTTP REST protocol. *TODO: we need to have a discussion about security here: TLS, authn/authz*
+
+**`GetStatus GET /v3/status`**
```

Comment:
From a sidebar discussion @misterbisson and I had today:

- A read-only status endpoint like this path can be moved under the telemetry interface, under a different path. This lets us use the same listener (and config) for the Prometheus metrics.
- The rest of the control plane can just listen on a unix domain socket (at least for the initial implementation). This means it's not exposed to the network and can be made reachable only to ContainerPilot hooks without having to implement TLS or authn/authz